### PR TITLE
Adjustments for split read and write DB

### PR DESCRIPTION
**Pull request by pseudonom**
Goes with https://github.com/yesodweb/persistent/pull/534

### CODE REVIEW

**Comment by pseudonom** [comment]
Hm, does this permission denied error in travis make sense to anyone? Is it a transient thing that triggering a rebuild might solve?
```
Cloning into '/home/travis/build/yesodweb/yesod/.stack-work/downloaded/338de38a6fac895a0dfdfd64cd279e13cc4286a72fd4ed186255fe57a7ebe3c1.tmp'...
Warning: Permanently added the RSA host key for IP address '192.30.252.131' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by snoyberg** [comment]
Do you have a submodule using ssh auth?

**Comment by pseudonom** [comment]
Ah, I did add a git reference in `stack.yaml` to point to the PRed version of persistent.

```yml
 - location:
      git: git@github.com:pseudonom/persistent.git
      commit: 0edba747e49c816e94c235bced16978cb729b175
    subdirs:
    - persistent
```

Is there a way to make this work with travis or should I just temporarily vendor that version of `persistent` in a subdirectory?

**Comment by snoyberg** [comment]
Just switch to the https URL instead, it should work just fine.

**Comment by pseudonom** [comment]
Ah, thanks! Didn't realize `stack` supported both. Looks like everything should be good to go now?

**Comment by pseudonom** [comment]
Bumped it to work with the newly released `persistent-2.5`.

File: yesod-auth/Yesod/Auth.hs [hunk 1]
```diff
@@ -500,10 +500,10 @@ class (YesodAuth master, YesodPersist master) => YesodAuthPersist master where
     getAuthEntity :: AuthId master -> HandlerT master IO (Maybe (AuthEntity master))
 
     default getAuthEntity
-        :: ( YesodPersistBackend master
-               ~ PersistEntityBackend (AuthEntity master)
+        :: ( YesodPersistBackend master ~ backend
+           , BaseBackend backend ~ PersistEntityBackend (AuthEntity master)
```
Comment (gregwebs) [inline yesod-auth/Yesod/Auth.hs]: Can use the new constraint kind synonym here.

**Comment by gregwebs** [comment]
looks good other than it could use `PersistRecordBackend`. I will let @snoyberg merge this.

**Comment by snoyberg** [comment]
To avoid upgrade headaches, we should keep compatibility with the old version of persistent via CPP. It's a pain, but it's better to keep the pain centralized in Yesod itself and not affect users. Would you be able to update the PR?

**Comment by pseudonom** [comment]
Sure, I can do that.

**Comment by pseudonom** [comment]
Okay, `CPP`ed it.

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**